pkg/hubble/parser/threefour/parser.go

```diff
@@ -481,6 +481,7 @@ func decodeTrafficDirection(srcEP uint32, dn *monitor.DropNotify, tn *monitor.Tr
 
 			// isSourceEP != isReply ==
 			//  (isSourceEP && !isReply) || (!isSourceEP && isReply)
+			// FIXME: currently broken for monitor.TraceReasonEncryptOverlay showing INGRESS
```

Comment:
Would you mind filing an issue for this? Elsewhere in Cilium we often use the format like:

```suggestion
			// GH-XXXX: currently broken for monitor.TraceReasonEncryptOverlay showing INGRESS
```

This just makes it easier to (a) track the fix, (b) attract someone to fix it, (c) conduct any discussion about solutions.

Reply:
Actually the discussion point I raised immediately below may render this point unnecessary.

Something else I note though is that `TraceReasonEncryptOverlay` is _probably_ showing up as INGRESS because the packet is INGRESSing the stack right? I don't know if we've 100% locked down the meaning of INGRESS/EGRESS but if this is effectively the Linux stack perspective, ie TC ingress/egress then it may even be accurate. Confusing for users, but somewhat consistent with existing flows.

Reply:
@kaworu I think we are already tracking an issue for this, no?  

I made the same comment to Alex. Ingress is actually correct here, but it may not be correct or appropriate when the flow is user facing in hubble. 

I think this should become more of a conversation here: https://github.com/cilium/cilium/pull/31226 or in a follow up issue as I don't believe these semantics are really a focus of this PR.

Reply:
Leaving this open until we determine the right gh-ticket to add to the comment above.

Reply:
> @kaworu I think we are already tracking an issue for this, no?

No issue yet, I'm intending to address it as part of https://github.com/cilium/cilium/pull/31226.

> [...] but somewhat consistent with existing flows.

@joestringer As you mentioned in another thread on this PR, the packet will be processed later and generate another Hubble flow that will be shown as EGRESS (which at this point will make complete sense). For this trace notification, while INGRESS can be seen technically correct, I don't think having the same packet generating an INGRESS _and_ an EGRESS flow make sense overall at a high level (e.g. metrics accounting). Thus my plan is basically to consistently set `traffic_direction=EGRESS` on Hubble flows generated from a `TraceReasonEncryptOverlay` notification.

Reply:
Cool, at the minimum I will update the comment in the code to point to #31226 :)

Reply:
Commented updated, resolving.